### PR TITLE
Improve filer keybindings

### LINKED ISSUE #521: filer: <CR> on directory should work like <TAB> and change to dir instead of calling :Edit on a directory
At least configurable if anyone sees any good in current behaviour.

### PR DESCRIPTION
**Pull request by romgrk**
Fixes #521 

This PR improves filer keybindings as described in the issue above as follow:
 - `<CR>` on a directory changes filer's directory to that directory
 - `<CR>` on a file opens the file for editing
 - `<A-u>` (mnemonic: `alt-Up`) at any point goes to the parent directory

`<A-u>` is better than going to the parent directory on `<BS>` because the user might realize mid-input that the file they're looking for isn't there, and want to go back. Needing to erase the whole input before isn't ideal.

To implement the changes above, this PR makes these changes:
 - `<CR>` nows has `clap#handler#cr_action` instead of calling directly `clap#handler#sink`. Calls `provider.cr_action` or `clap#handler#sink`.
 - `<A-u>` is the default mapping for a new concept: `back_action`. It is optionally provided as `provider.back_action`. In filer, it correspond to go up, but it can have other meanings in other providers.

### CODE REVIEW

**Comment by liuchengxu** [comment]
LGTM. @Grueslayer you could try this PR :)

**Comment by romgrk** [comment]
Todo before merging:
 - [ ] Update documentation on providers
 - [ ] Update changelog

**Comment by Grueslayer** [comment]
> LGTM. @Grueslayer you could try this PR :)

Works fine for me 👍

**Comment by romgrk** [comment]
All done on my side, you can merge the PR if you're ok with it.

File: README.md [hunk 1]
```diff
@@ -219,6 +219,8 @@ See `:help clap-options` for more information.
 - [x] Use <kbd>Enter</kbd> to select the entry and exit.
```
Comment (liuchengxu) [inline README.md:219]:
```suggestion
- [x] Use <kbd>Enter</kbd> to select the entry and exit.
  -  Use <kbd>Enter</kbd> to expand the directory or edit the file for `:Clap filer`.
- [x] By default <kbd>Alt-u</kbd> does nothing.
  - Use <kbd>Alt-u</kbd> to go up one directory in `:Clap filer`.
```

Comment (romgrk) [reply README.md:219]: Applied

File: README.md [hunk 2]
```diff
@@ -217,8 +217,13 @@ See `:help clap-options` for more information.
 - [x] Use <kbd>Ctrl-b</kbd> to move cursor left one character.
 - [x] Use <kbd>Ctrl-f</kbd> to move cursor right one character.
 - [x] Use <kbd>Enter</kbd> to select the entry and exit.
+  -  Use <kbd>Enter</kbd> to expand the directory or edit the file for `:Clap filer`.
+- [x] By default <kbd>Alt-u</kbd> does nothing.
+  - Use <kbd>Alt-u</kbd> to go up one directory in `:Clap filer`.
 - [x] Use <kbd>Tab</kbd> to select multiple entries and open them using the quickfix window.(Need the provider has `sink*` support)
   - Use <kbd>Tab</kbd> to expand the directory for `:Clap filer`.
+- [x] Use <kbd>Enter</kbd> to expand the directory or edit the file for `:Clap filer`.
+- [x] Use <kbd>Alt-u</kbd> to go up one directory in `:Clap filer`.
```
Comment (liuchengxu) [inline README.md:226]:
```suggestion
```

**Comment by liuchengxu** [comment]
Merged, thank you!

**Pull request merged**